src/redux/compare/state.ts

```diff
@@ -0,0 +1,143 @@
+import {K8sResource} from '@models/k8sresource';
+
+export const initialState: CompareState = {
+  isOpen: false,
+  views: [],
+  current: {
+    view: {
+      operation: 'union',
+      leftSet: undefined,
+      rightSet: undefined,
+    },
+    selection: [],
+    transfering: {
+      pending: false,
+    },
+  },
+};
+
+export type CompareState = {
```

Comment:
I would add comments to the fields to explain how/why they are used, similar to how we do in `models/k8sresource.ts`